test/powershell/engine/Module/ModuleCmdlet.Tests.ps1

```diff
@@ -0,0 +1,506 @@
+            $module.'Test-SecondModuleFunction'() | Should -BeExactly "SECONDSTRING"
+        }
+    }
+
+    Context "Classes in modules" {
+        It "Uses updated class definitions when the module is reloaded" {
+            $content = @'
+$passedArgs = $Args
+class Root { $passedArgs = $passedArgs }
+function Get-PassedArgsRoot { [Root]::new().passedArgs }
+function Get-PassedArgsNoRoot { $passedArgs }
+'@
+
+            $rootModName = "rootMod"
+            $rootModDir = "$TestDrive\$rootModName"
+            New-Item -Path $rootModDir -ItemType Directory
+            New-Item -Path $rootModDir -Name "$rootModName.psm1" -Value $content
+
+            Import-Module $rootModDir -ArgumentList 'value1'
+            $rootVal = Get-PassedArgsRoot
+            $noRootVal = Get-PassedArgsNoRoot
+            $rootVal | Should -BeExactly $noRootVal
+
+            Import-Module $rootModDir -ArgumentList 'value2'
+            $rootVal = Get-PassedArgsRoot
+            $rootNoVal = Get-PassedArgsNoRoot
+            $rootVal | Should -BeExactly $noRootVal
+        }
+
+    }
+
+    Context "Parameter handling" {
+        BeforeAll {
+            $validationTests = @(
+                @{ paramName="Variable" },
+                @{ paramName="Function" },
+                @{ paramName="Cmdlet" },
+                @{ paramName="Variable" }
+            )
+
+            $ps = [powershell]::Create()
+        }
+
+        AfterAll {
+            $ps.Dispose()
+        }
+
+        AfterEach {
+            $ps.Streams.ClearStreams()
+        }
+
+        It "Validates a null -<paramName> parameter" -TestCases $validationTests {
+            param($paramName)
+
+            $modItem = New-SimpleModule
+            try
+            {
+                $ps.AddScript("Import-Module $($modItem.PSPath) -$paramName `$null")
+                $ps.Invoke()
+            }
+            catch
+            {
+                $_.FullyQualifiedErrorId | Should -BeExactly "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.ImportModuleCommand"
+            }
+        }
+
+        It "Ignores whitespace -MaximumVersion parameter" {
+            $modItem = New-SimpleModule
+            Import-Module $modItem.PSPath -MaximumVersion "    "
+            Test-FirstModuleFunction | Should -BeExactly "TESTSTRING"
+        }
+    }
+}
+
+Describe "Import-Module by PSModuleInfo" -Tag Feature {
+    BeforeAll {
+        $modPath = (New-SimpleModule).PSPath
+        $modRef = Import-Module $modPath -PassThru
+    }
+
+    BeforeEach {
+        Get-Module $modRef.Name | Remove-Module
+        $modItem = New-SimpleModule
+        $modInfo = (Get-Module $modItem.PSPath -ListAvailable)[0]
+    }
+
+    It "Imports a module by moduleinfo" {
+        try
+        {
+            Test-FirstModuleFunction
+        }
+        catch
+        {
+            $_.Exception | Should -BeOfType [System.Management.Automation.CommandNotFoundException]
```

Comment:
What do the test check? Exception? If so we should use  `{ ... } | Should -Throw -ErrorId " ... "` Also we could add `-Throu` to get the exception.